source/macros.tex

```diff
@@ -191,6 +191,9 @@
 \newcommand{\indexlibraryglobal}[1]{\indexlibrary{\idxcode{#1}}}
 \newcommand{\indexlibraryctor}[1]{\indexlibrary{\idxcode{#1}!constructor}}
 \newcommand{\indexlibrarydtor}[1]{\indexlibrary{\idxcode{#1}!destructor}}
+\newcommand{\indexlibraryspec}[2]{\indexlibrary{\idxcode{#1}!\idxcode{#2}}}
+\def\deflibspec#1<#2<#3>{\tcode{#1<#2<#3>}\indexlibraryspec{#1}{#2}} % partial specialization
+\def\deflibespecq#1<#2::#3>{\tcode{#1<#2::#3>}\indexlibraryspec{#1}{#3}} % explicit specialization of qualified-id
```

Comment:
```suggestion
\newcommand{\indexlibraryspec}[2]{\indexlibrary{\idxcode{#1}!\idxcode{#2}}}
\newcommand{\deflibspec}[3]{\tcode{#1<#2#3}\indexlibraryspec{#1}{#3}}
```
Would this be an improvement?
Example uses:
```LaTeX
    inline constexpr bool ranges::@\deflibspec{enable_view}{}{span}@<ElementType, Extent>> = true;
    inline constexpr bool @\deflibspec{enable_borrowed_range}{filesystem::}{directory_iterator}@> = true;
    inline constexpr bool @\deflibspec{disable_sized_sentinel_for}{}{reverse_iterator}@<Iterator1>,
                                                     reverse_iterator<Iterator2>> = true;
```

Reply:
Alternatively, rather than inserting
```LaTeX
\indexlibraryspec{enable_borrowed_range}{reverse_view}%
```
before a code block, do it inline before the specialization, assuming that's possible:
```LaTeX
    @\indexlibraryspec{enable_view}{span}@
    inline constexpr bool ranges::enable_view<span<ElementType, Extent>> = true;
```
Or define `\deflibspec` like the adjective definition commands, which have some repetition: `\defnadj{a}{b}{b!a}`:
```LaTeX
    inline constexpr bool ranges::@\deflibspec{enable_view}{span}@<span<ElementType, Extent>> = true;
    inline constexpr bool @\deflibspec{enable_borrowed_range}{directory_iterator}@<filesystem::directory_iterator> = true;
    inline constexpr bool @\deflibspec{disable_sized_sentinel_for}{reverse_iterator}@<reverse_iterator<Iterator1>,
                                                     reverse_iterator<Iterator2>> = true;
```